app/lib/admin/models.dart

```diff
@@ -196,3 +233,35 @@ class ModerationSubjectKind {
   static const packageVersion = 'package-version';
   static const publisher = 'publisher';
 }
+
+@JsonSerializable(includeIfNull: false)
+class ModerationActionLog {
+  final List<ModerationActionLogEntry> entries;
```

Comment:
```suggestion
  final List<({DateTime timestamp, String subject, bool applyModeration, bool revertModeration, String message})> entries;
```

How about using records? If it works with json_serializable :see_no_evil:

Reply:
What's the benefit of records here? I think this is more than just a multi-type return value.

Reply:
`json_serializable` does support this, however, I still think it is better to keep it in a separate class.

Also: added `message` field, but I'm undecided on the  `applyModeration` / `revertModeration` fields. How about the `enum ModerationAction {apply, revert}`?

Reply:
I don't have a problem with an enum